contracts/index_executor/src/lib.rs

```diff
@@ -210,35 +290,37 @@ mod index_executor {
 
                 // Submit to blockchain
                 if let Some(submittable) = maybe_submittable {
-                    let _tx_id = submittable
+                    let tx_id = submittable
                         .submit(&self.executor_account, 0)
                         .log_err("failed to submit rollup tx")
                         .or(Err(Error::FailedToSendTransaction))?;
+                    pink_extension::debug!(
+                        "Send transaction to set worker account: {:?}",
+                        hex::encode(tx_id)
+                    );
                 }
             }
-
+            Self::env().emit_event(WorkerSetToRollup {});
             Ok(())
         }
 
         #[ink(message)]
         pub fn run(&self, running_type: RunningType) -> Result<()> {
             let config = self.ensure_configured()?;
-            // Get rpc info from registry
-            let chain = self
-                .get_chain("Khala".to_string())
-                .map(Ok)
-                .unwrap_or(Err(Error::ChainNotFound))?;
             let contract_id = self.env().account_id();
             let mut client = SubstrateRollupClient::new(
-                &chain.endpoint,
-                config.pallet_id.unwrap(),
+                &config.rollup_endpoint,
+                config.rollup_pallet_id,
                 &contract_id,
+                SUB_ROLLUP_PREFIX,
             )
             .log_err("failed to create rollup client")
             .or(Err(Error::FailedToCreateClient))?;
 
             match running_type {
-                RunningType::Fetch(source_chain) => self.fetch_task(&mut client, source_chain)?,
+                RunningType::Fetch(source_chain, worker) => {
```

Comment:
It's better to have a backend so it can work `singleton`. And it's ok if we distribute a running worker to the task on client, we have a queue in `handler` contract